permission_handler_platform_interface/lib/src/permission_status.dart

```diff
@@ -12,19 +12,34 @@ enum PermissionStatus {
   /// The OS denied access to the requested feature. The user cannot change
   /// this app's status, possibly due to active restrictions such as parental
   /// controls being in place.
+  ///
   /// *Only supported on iOS.*
   restricted,
 
-  ///User has authorized this application for limited access.
+  /// The user has authorized this application for limited access. So far this
+  /// is only relevant for the Photo Library picker.
+  ///
   /// *Only supported on iOS (iOS14+).*
   limited,
 
   /// Permission to the requested feature is permanently denied, the permission
   /// dialog will not be shown when requesting this permission. The user may
   /// still change the permission status in the settings.
+  ///
+  /// *On Android:*
+  /// Android 11+ (API 30+): whether the user denied the permission for a second
+  /// time.
+  /// Below Android 11 (API 30): whether the user denied access to the requested
+  /// feature and selected to never again show a request.
+  ///
+  /// *On iOS:*
+  /// If the user has denied access to the requested feature.
+  ///
+  /// WARNING: This can only be determined AFTER requesting this permission.
+  /// Therefore make a `request` call first.
```

Comment:
Since version 11.0.0 this is no longer relevant, we have added some logic that would cache the permanentlyDenied permission and will also return it when calling `checkPermission`.

Reply:
ack.